### ISSUE TITLE
When to play hls stream? After ts has been downloaded or playing during downloading

### ISSUE THREAD

**Issue by sl950313** [google/ExoPlayer#5558]
Hi, I wonder to know when a ts file will be extracted and played by the exoplayer. as i know the hls.js will extract and play ts after the file downloads from server,  but I see the log 

```
02-22 18:11:52.253 30427-30712/com.google.android.exoplayer2.demo I/ExtendedACodec: Decoder will be in frame by frame mode
```
What does the log mean ?

**Comment by andrewlewis** [comment]
During playback of HLS streams we extract data from media segments while they're downloading. That log line is probably not relevant to this.

**Comment by sl950313** [comment]
> During playback of HLS streams we extract data from media segments while they're downloading. That log line is probably not relevant to this.

Thank you for help. But what does media segments mean? not frame by frame?

**Comment by andrewlewis** [comment]
They're pieces of media (typically a few seconds long). As the log line above doesn't seem to indicate a problem I will close this issue. If your issue isn't actually resolved, please could you rephrase what the problem is? Thanks.
